src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/ValidateAnomalyDetectorTransportAction.java

```diff
@@ -0,0 +1,226 @@
+            request.getDetector(),
+            request.getRequestTimeout(),
+            request.getMaxSingleEntityAnomalyDetectors(),
+            request.getMaxMultiEntityAnomalyDetectors(),
+            request.getMaxAnomalyFeatures(),
+            RestRequest.Method.POST,
+            xContentRegistry,
+            user,
+            searchFeatureDao,
+            request.getTypeStr()
+        );
+        try {
+            handler.start();
+        } catch (Exception exception) {
+            String errorMessage = String
+                .format(Locale.ROOT, "Unknown exception caught while validating detector %s", request.getDetector());
+            logger.error(errorMessage, exception);
+            listener.onFailure(exception);
+        }
+    }
+
+    protected DetectorValidationIssue parseADValidationException(ADValidationException exception) {
+        String originalErrorMessage = exception.getMessage();
+        String errorMessage;
+        Map<String, String> subIssues = null;
+        Object suggestion = null;
+        switch (exception.getType()) {
+            case FEATURE_ATTRIBUTES:
+                logger.info("originalErrorMessage: " + originalErrorMessage);
+                // feature issues will be between square brackets like
+                // [Feature has issue: A, Feature has issue: B]
+                int firstLeftBracketIndex = originalErrorMessage.indexOf("[");
+                int firstRightBracketIndex = originalErrorMessage.lastIndexOf("]");
+                if (firstLeftBracketIndex != -1 && firstRightBracketIndex != -1) {
+                    errorMessage = originalErrorMessage.substring(firstLeftBracketIndex + 1, firstRightBracketIndex);
+                    logger.info("errorMessage: " + errorMessage);
+                    subIssues = getFeatureSubIssuesFromErrorMessage(errorMessage);
+                } else {
+                    errorMessage = originalErrorMessage;
+                }
+                break;
+            case NAME:
+            case CATEGORY:
+            case DETECTION_INTERVAL:
+            case FILTER_QUERY:
+            case TIMEFIELD_FIELD:
+            case SHINGLE_SIZE_FIELD:
+            case INDICES:
+                errorMessage = originalErrorMessage;
+                break;
+            default:
+                logger.warn(String.format(Locale.ROOT, "Invalid AD validation exception type: %s", exception));
+                return null;
+        }
+        return new DetectorValidationIssue(exception.getAspect(), exception.getType(), errorMessage, subIssues, suggestion);
+    }
+
+    private Map<String, String> getFeatureSubIssuesFromErrorMessage(String errorMessage) {
+        Map<String, String> result = new HashMap<String, String>();
+        String featureSubIssueMessageSeparator = ", " + FEATURE_INVALID_MSG_PREFIX;
+        // search backwards from end
+        int foundSubIssueIndex = errorMessage.lastIndexOf(featureSubIssueMessageSeparator);
```

Comment:
Can we simplify with `errorMessage.split(featureSubIssueMessageSeparator)`

Reply:
sure. will do that.